src/operun/crm/ldap/ldap_sync.py

```diff
@@ -0,0 +1,289 @@
+# -*- coding: utf-8 -*-
+"""
+LDAP Sync View & Utils.
+"""
+
+from operun.crm import MessageFactory as _
+from plone import api
+from Products.Five.browser import BrowserView
+from Products.Five.browser.pagetemplatefile import ViewPageTemplateFile
+
+import ldap
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+class LdapSyncView(BrowserView):
+
+    template = ViewPageTemplateFile('templates/ldap_sync.pt')
+
+    def __call__(self):
+        return self.template()
+
+    # LDAP Object Methods
+
+    def add_ldap_object(self, item=None):
+        """
+        Create object in LDAP with content_type and UID.
+        Get list of fields config using get_field_mapping method.
+        Iterate over list and update each attribute.
+        """
+        connection = self.connect()
+        if connection:
+            if not item:
+                item = self.context
+            mod_attrs = self.generate_create_mod_attrs(item)
```

Comment:
Please turn this around. First set rename item to object because we expect an object. I think this can be done for the whole view. Set obj = self.context so context is default.